pandas/core/frame.py

```diff
@@ -695,8 +695,9 @@ def iterrows(self):
 
         """
         columns = self.columns
+        klass = self._constructor_sliced
         for k, v in zip(self.index, self.values):
-            s = Series(v, index=columns, name=k)
+            s = klass(v, index=columns, name=k)
```

Comment:
I took existing code as example and `__finalize__` is never called for `_constuctor_sliced`. :worried: